testing/integration/agent_long_running_leak_test.go

```diff
@@ -0,0 +1,296 @@
+	Handles HandlesMetrics `mapstructure:"handles"`
+}
+
+type processWatcher struct {
+	handle     types.Process
+	pid        int
+	name       string
+	regHandles tools.Slope
+}
+
+func TestLongRunningAgentForLeaks(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{Type: define.Linux},
+			{Type: define.Windows},
+		},
+	})
+
+	if os.Getenv("TEST_LONG_RUNNING") == "" {
+		t.Skipf("not running extended test unless TEST_LONG_RUNNING is set")
+	}
+
+	suite.Run(t, &ExtendedRunner{info: info})
+}
+
+func (runner *ExtendedRunner) SetupSuite() {
+	cmd := exec.Command("go", "install", "-v", "github.com/mingrammer/flog@latest")
+	out, err := cmd.CombinedOutput()
+	require.NoError(runner.T(), err, "got out: %s", string(out))
+
+	cmd = exec.Command("flog", "-t", "log", "-f", "apache_error", "-o", "/var/log/httpd/error_log", "-b", "50485760", "-p", "1048576")
+	out, err = cmd.CombinedOutput()
+	require.NoError(runner.T(), err, "got out: %s", string(out))
+
+	policyUUID := uuid.New().String()
+	unpr := false
+	installOpts := atesting.InstallOpts{
+		NonInteractive: true,
+		Force:          true,
+		Unprivileged:   &unpr,
+	}
+
+	fixture, err := define.NewFixture(runner.T(), define.Version())
+	require.NoError(runner.T(), err)
+	runner.agentFixture = fixture
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancel()
+
+	basePolicy := kibana.AgentPolicy{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+
+	// install system package
+	runner.InstallPackage(ctx, "system", "1.53.1", "agent_long_test_base_system_integ.json", uuid.New().String(), policyResp.ID)
+
+	// install cef
+	runner.InstallPackage(ctx, "apache", "1.17.0", "agent_long_test_apache.json", uuid.New().String(), policyResp.ID)
+
+}
+
+func (runner *ExtendedRunner) InstallPackage(ctx context.Context, name string, version string, cfgFile string, policyUUID string, policyID string) {
+	installPackage := kibana.PackagePolicyRequest{}
+
+	jsonRaw, err := os.ReadFile(cfgFile)
+	require.NoError(runner.T(), err)
+
+	err = json.Unmarshal(jsonRaw, &installPackage)
+	require.NoError(runner.T(), err)
+
+	installPackage.Package.Version = version
+	installPackage.ID = policyUUID
+	installPackage.PolicyID = policyID
+	installPackage.Namespace = "default"
+	installPackage.Name = fmt.Sprintf("%s-long-test-%s", name, policyUUID)
+	installPackage.Vars = map[string]interface{}{}
+
+	runner.T().Logf("Installing %s package....", name)
```

Comment:
this function doesn't need to be part of the fixture just to get `runner.T()` we can pass `t *testing.T` as parameter

Reply:
It does use the runners to fetch the kibana info. Figured that was cleaner than passing a few extra args.